File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.cpp
bool compare(vector<int>& a, vector<int>& b) {
    return a[1] < b[1];
}
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), compare);
        int ans = 0;
        int k = INT_MIN;

        for (int i = 0; i < intervals.size(); i++) {
            int x = intervals[i][0];
            int y = intervals[i][1];

            if (x >= k) {
                k = y;
            } else {
                ans++;
            }
        } 
        return ans;
    }
};

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        from heapq import heappop, heappush
        from collections import deque
        start = []
        end = []
        for s, e in intervals:
            start.append(s)
            end.append(e)
        start.sort()
        end.sort()
        start = deque(start)
        end = deque(end)
        dup, max_dup = 0, 0
        s = start.popleft()
        e = end.popleft()
        while s >= -5e4 and e >= -5e4:
            if s < e:
                dup += 1
                max_dup = max(max_dup, dup) 
                if not start:
                    return max_dup - 1
                s = start.popleft()
            if e < s:
                dup -= 1
                e = end.popleft()
            if e == s:
                if start:
                    s = start.popleft()
                    e = end.popleft()
                else:
                    return max_dup - 1
